<?xml version="1.0"?>

<!-- Launch file for the starting mission for the thesis of Boris Dehem-->
<!-- Work In Progress -->
<launch>
  <arg name="threshold"              default="250" />
  <arg name="max_matches"            default="200"/>
  <arg name="no_bundle_adjustment"   default="false"/>
  <arg name="dlt_triangulation"      default="false"/>
  <arg name="midpoint_triangulation" default="false"/>
  <arg name="test_robustness"        default="false"/>
  <arg name="double_ba"              default="false"/>
  <arg name="bundle_adjustment_tol"  default="0.005"/>
  <arg name="quiet_ba"               default="false" />
  <arg name="rpt2"                   default="6" />
  <arg name="rpt3"                   default="-1" />
  <arg name="rpt4"                   default="-1" />
  <arg name="only_init"              default="false" />
  <arg name="remove_cst"             default="1" />
  <arg name="remove_coeff"           default="3" />
  <arg name="huber_delta"            default="1.0" />
  <arg name="manual_keyframes"       default="false" />
  <arg name="benchmark"              default="false" />

  <arg name="min_dist"        default="0.3" />
  <arg name="min_time"        default="2" />
  <arg name="time_thresh"     default="5" />
  <arg name="inliers_thresh"  default="40" />
  <arg name="n_kf_local_ba"   default="6" />
  <arg name="freq_global_ba"  default="5" />

  <node name="image_proc_front" pkg="image_proc" type="image_proc" ns="ardrone/front"/>

  <node name="boris_drone_mapping_node" pkg="boris_drone" type="mapping_node" output="screen">
    <param name="test_robustness"        value="$(arg test_robustness)" />
    <param name="double_ba"              value="$(arg double_ba)"/>
    <param name="threshold_kf_match"     value="$(arg threshold)" />
    <param name="max_matches"            value="$(arg max_matches)" />
    <param name="no_bundle_adjustment"   value="$(arg no_bundle_adjustment)"/>
    <param name="dlt_triangulation"      value="$(arg dlt_triangulation)"/>
    <param name="midpoint_triangulation" value="$(arg midpoint_triangulation)"/>
    <param name="only_init"              value="$(arg only_init)"/>
    <param name="rpt2"                   value="$(arg rpt2)"/>
    <param name="rpt3"                   value="$(arg rpt3)"/>
    <param name="rpt4"                   value="$(arg rpt4)"/>
    <param name="remove_cst"             value="$(arg remove_cst)"/>
    <param name="remove_coeff"           value="$(arg remove_coeff)"/>
    <param name="manual_keyframes"       value="$(arg manual_keyframes)"/>
    <param name="benchmark"              value="$(arg benchmark)" />
    <param name="only_init"              value="true" />

    <param name="min_dist"        value="$(arg min_dist)" />
    <param name="min_time"        value="$(arg min_time)" />
    <param name="time_thresh"     value="$(arg time_thresh)" />
    <param name="inliers_thresh"  value="$(arg inliers_thresh)" />
    <param name="n_kf_local_ba"   value="$(arg n_kf_local_ba)" />
    <param name="freq_global_ba"  value="$(arg freq_global_ba)" />
  </node>

  <node name="boris_drone_bundle_adjuster" pkg="boris_drone" type="bundle_adjuster" output="screen">
    <param name="bundle_adjustment_tol" value="$(arg bundle_adjustment_tol)" />
    <param name="quiet"                 value="$(arg quiet_ba)" />
    <param name="huber_delta"           value="$(arg huber_delta)"/>
  </node>
</launch>
